src/lakefs_spec/spec.py

```diff
@@ -486,6 +486,7 @@ def ls(
             A list of all objects' metadata under the given remote path if ``detail=True``, or alternatively only their names if ``detail=False``.
         """
         path = stringify_path(path)
+        path = self._strip_protocol(path)
```

Comment:
`stringify_path()` is implied by `_strip_protocol()`, so you can replace L488 entirely.